system/core/pages.php

```diff
@@ -276,18 +277,18 @@ function loadPage(RequestContext $O_O) {
 	registerCommandCallback('newPage', 'newPage');
 	function newPage($newName) {
 		global $hyphaXml, $hyphaUrl, $hyphaContentLanguage;
-
+		$language = array_key_exists('newPageLanguage', $_POST) ? $_POST['newPageLanguage'] : $hyphaContentLanguage;
```

Comment:
I think this should validate the language code, since AFAICS `hypha_addPage` just uses it as-is.

Reply:
fixed at `hypha_addPage `